src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -23,16 +23,46 @@
 from deepsparse.pipeline import create_engine
 from deepsparse.transformers.utils.decoder_kv_cache import DecoderKVCache
 from deepsparse.transformers.utils.helpers import generate_session_id, softmax
+from deepsparse.transformers.utils.storage_kv_cache import KVCacheSessionStorage
 from sparsezoo.utils.onnx import save_onnx
 
 
 _LOGGER = logging.getLogger(__name__)
 
-__all__ = ["NLDecoderEngine"]
+__all__ = ["NLDecoderEngine", "synchronise_engines_cache"]
 
 _CACHE_INPUT_NAME = "past_key_values"
 
 
+def synchronise_engines_cache(engines: Set["NLDecoderEngine"]):  # noqa F821
```

Comment:
What is the desired use case for this? Why do we want to synchronize the *entire* caches between pipelines rather than just the individual cache of the session that should be transferred

Reply:
Agree, changes introduced